src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -83,6 +83,25 @@ class ExecutionStateProgramCounterBinder {
     size_t* m_oldAddress;
 };
 
+template <typename T>
+class ExecutionStateVariableChanger {
+public:
+    ExecutionStateVariableChanger(ExecutionState& state, T changer)
+        : m_state(state)
+        , m_changer(changer)
+    {
+        m_changer(m_state, true);
+    }
+    ~ExecutionStateVariableChanger()
+    {
+        m_changer(m_state, false);
+    }
+
+private:
+    ExecutionState& m_state;
+    T m_changer;
```

Comment:
How about using `std::function` as suggested below?

```suggestion
    ExecutionStateVariableChanger(std::function<T> changer)
        : m_changer(changer)
    {
        m_changer(true);
    }
    ~ExecutionStateVariableChanger()
    {
        m_changer(false);
    }

private:
    std::function<T> m_changer;
```

As a result, it can provide more general usability and shorter code like below
```c++
ExecutionStateVariableChanger<void(bool)> changer([state](bool in) {
    state->m_onTry = in;
});
```

Reply:
using capture with std::function is expensive(It needs malloc/free when using lamba capture)
that's why I use function pointer

Reply:
even you don't use capture, std::function make code shorter but it is not required suggestion :)

Reply:
I don't want to make capture usable.
BTW std::function use malloc/free and we cannot change allocator of lambda.
so It is not traced by bdwgc. It's dangerous